packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -533,14 +539,25 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     if (typeof newChild === 'object' && newChild !== null) {
       switch (newChild.$$typeof) {
         case REACT_ELEMENT_TYPE: {
-          const created = createFiberFromElement(
-            newChild,
-            returnFiber.internalContextTag,
-            expirationTime,
-          );
-          created.ref = coerceRef(null, newChild);
-          created.return = returnFiber;
-          return created;
+          if (newChild.type === REACT_FRAGMENT_TYPE) {
+            const created = createFiberFromFragment(
+              newChild.props.children,
+              returnFiber.internalContextTag,
+              expirationTime,
+              newChild.key,
+            );
```

Comment:
I wonder if we should have a warning for `ref` on a fragment, either in ReactChildFiber or in ReactElementValidator?

Reply:
Oops, I assumed that `ref` would appear in `element.props`. I think it makes sense to do the check in ReactElementValidator since that's the earliest possible time we can catch the potential bug right? What are the advantages of having the warning inside ReactChildFiber?

Reply:
I think other ref warnings are in the reconciler. The benefit of that is that we (or others) can create changes to the reconciler (such as adding ref support on fragments) without changing the isomorphic package. Another issue is that people can create elements in other ways (e.g. by inlining) and not get the warning.

But as you said, it's nice to get it earlier for stack traces.

I don't really have a preference.

Reply:
When you say that people can create elements by inlining, this can only happen in non-dev mode right? (I'm assuming that when you mention inlining, you are referring to this transform: https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-react-inline-elements)